### PR TITLE
Feature/33963 Handle D365 payload data truncation

### PR DESCRIPTION
**Pull request by vipin14904**
Feature/33963 new d365 payload data truncation.

### CODE REVIEW

**Review by usha13529 — APPROVED** [review]
LGTM.

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/SubscriptionController.cs [hunk 1]
```diff
@@ -32,6 +33,11 @@ public virtual async Task<IActionResult> Post([FromBody] D365Payload d365Payload
         {
             _logger.LogInformation(EventIds.ENSSubscriptionRequestStart.ToEventId(), "Subscription request for D365Payload:{d365Payload} with _X-Correlation-ID:{correlationId}", JsonConvert.SerializeObject(d365Payload), GetCurrentCorrelationId());
 
+            if (HttpContext.Request.Headers.ContainsKey(_xmsDynamicsMsgSizeExceededHeader))
```
Comment (barrie-cooper) [inline UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/SubscriptionController.cs:36]: As above:
```suggestion
            if (HttpContext.Request.Headers.ContainsKey(XMsDynamicsMsgSizeExceededHeader))
```

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/SubscriptionController.cs [hunk 2]
```diff
@@ -20,6 +20,7 @@ public class SubscriptionController : BaseController<SubscriptionController>
         private readonly ILogger<SubscriptionController> _logger;
         private readonly ISubscriptionService _subscriptionService;
         private List<Error> _errors;
+        private readonly string _xmsDynamicsMsgSizeExceededHeader = "x-ms-dynamics-msg-size-exceeded";
```
[the remark this answers is not in the document]
Comment (vipin14904) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/SubscriptionController.cs:23]: Changes done as requested

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/SubscriptionController.cs (hunk 1 above)
Comment (vipin14904) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/SubscriptionController.cs:36]: Changes done as requested

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Controllers/SubscriptionControllerTests.cs [hunk 3]
```diff
@@ -25,6 +25,7 @@ public class SubscriptionControllerTests
         private ISubscriptionService _fakeSubscriptionService;
         private D365Payload _fakeD365PayloadDetails;
         private SubscriptionRequest _fakeSubscriptionRequest;
+        private readonly string _xmsDynamicsMsgSizeExceededHeader = "x-ms-dynamics-msg-size-exceeded";
```
[the remark this answers is not in the document]
Comment (vipin14904) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Controllers/SubscriptionControllerTests.cs:28]: change from readonly tot const

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Controllers/SubscriptionControllerTests.cs [hunk 4]
```diff
@@ -68,6 +67,20 @@ public async Task WhenPostInvalidNullInputParameters_ThenReceiveBadRequest()
             Assert.AreEqual("D365Payload InputParameters cannot be blank or null.", errors.Errors.Single().Description);
         }
 
+        [Test]
+        public async Task WhenD365HttpPayloadSizeExceeded_ThenLogError()
+        {
+            DefaultHttpContext defaultHttpContext = new();
+            defaultHttpContext.Request.Headers.Add(_xmsDynamicsMsgSizeExceededHeader, string.Empty);
```
[the remark this answers is not in the document]
Comment (vipin14904) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.UnitTests/Controllers/SubscriptionControllerTests.cs:74]: Changes are done as requested

**Review by usha13529 — APPROVED** [review]

**Review by barrie-cooper — APPROVED** [review]
Thanks for the little change @vipin14904

**Pull request merged**